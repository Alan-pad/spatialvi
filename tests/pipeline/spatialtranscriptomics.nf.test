nextflow_pipeline {
    name "Test Spatialtranscriptomics QC workflow"
    script "main.nf"
    tag "pipeline"

    test("Spatialtranscriptomics outputs") {
        when {
            params {
                outdir = "$outputDir"
            }
        }

        then {
            def softwareVersions = path("$outputDir/pipeline_info/software_versions.yml").yaml
            if (softwareVersions.containsKey("Workflow")) { softwareVersions.Workflow.remove("Nextflow") }

            assertAll(
                { assert workflow.success },
                { assert snapshot(UTILS.removeNextflowVersion("$outputDir")).match("software_versions") },
                { assert snapshot(path("$outputDir/Andersson_Nat_Gen_2021_CID4465_subsampled/data/st_adata_processed.h5ad")).match("anndata") },
                { assert snapshot(path("$outputDir/Andersson_Nat_Gen_2021_CID4465_subsampled/degs/st_spatial_de.csv")).match("spatial_de") },
                { assert snapshot(
                    path("$outputDir/Andersson_Nat_Gen_2021_CID4465_subsampled/reports/st_clustering.html"),
                    path("$outputDir/Andersson_Nat_Gen_2021_CID4465_subsampled/reports/st_qc_and_normalisation.html"),
                    path("$outputDir/Andersson_Nat_Gen_2021_CID4465_subsampled/reports/st_spatial_de.html")
                ).match("reports")}
            )
        }
    }
}
