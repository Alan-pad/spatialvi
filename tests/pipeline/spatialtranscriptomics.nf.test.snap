{
    "reports": {
        "content": [
            "st_clustering.html:md5,c0eef3dad596f1489d30a0172871168d",
            "st_qc_and_normalisation.html:md5,dc7cbe12e1048f6d913b222e57e0c534",
            "st_spatial_de.html:md5,b486defdc23bf1f0d2c71b16126cad8c"
        ],
        "timestamp": "2023-06-07T07:25:08+0000"
    },
    "anndata": {
        "content": [
            "st_adata_processed.h5ad:md5,48dea93087c1ad0f78bfd0e65d8ba9f2"
        ],
        "timestamp": "2023-06-07T07:25:08+0000"
    },
    "spatial_de": {
        "content": [
            "st_spatial_de.csv:md5,fd1b6818b859d0d7a58a62b6e26925ae"
        ],
        "timestamp": "2023-06-07T07:25:08+0000"
    }
}